### DISCUSSION TITLE
Feature request: `--format raw`

### DISCUSSION THREAD

**Discussion by nagya** [ReagentX/imessage-exporter#407] [Ideas]
This export format would produce an export directory that is a subset of the original directory, containing copies of those files in the original (with the same relative path and name) that imessage-exporter would use to produce the other export formats. Attachments are included if `--copy-method` is set to anything but `disabled`; however its value is otherwise not used, i.e. no attachment conversion happens in a raw export.

Hard links would be created instead of cloning if `--link` is specified (see #406).

This feature would enable the user to archive their conversations as of a particular point in time, without having to archive the whole iOS backup directory or the ~/Library/Messages directory. The benefit over the other export formats is that the raw format is invariant to future changes in how the other export formats are produced, and allows the user to re-export to other formats later with a newer version of imessage-exporter, taking advantage of any improvements and bug fixes in the HTML or text exporter.

In other words,

```
imessage-exporter --format raw --db-path ORIG --export-path EXPORT1 <other_options>
imessage-exporter --format html --db-path EXPORT1 --export path EXPORT2 <other_options>
```

would produce the same result at EXPORT2 as:

```
imessage-exporter --format html --db-path ORIG --export path EXPORT2 <other_options>
```

**Comment by ReagentX** [comment]
Why not just backup `~/Library/Messages` (or just the `chat.db` within)?

**Reply by nagya** [reply]
This feature would definitely be more useful for iOS and iPadOS backups.

The user may not have a Mac, may not use it for messaging, or their message db on their Mac may not be fully in sync with their iPhone. Without Messages in iCloud, when one starts using a Mac for messaging, only newly sent iMessages and SMS texts show up on the Mac, historical ones are not synced. I've also seen that when the Mac is offline for an extended period of time, then a portion of the message history that happened while it was offline is not properly synced to it.
